EIPS/eip-4512.md

```diff
@@ -0,0 +1,284 @@
+     * @param user the address of token user.
+     * @param index the index of user token list.
+     * @return a tokenId used by `user` at a given `index` of its token list.
+     */
+    function tokenOfUserByIndex(address user, uint256 index) external view returns (uint256 tokenId);
+
+    /**
+     * @dev get the total amount of tokens stored by the contract.
+     * @return the total amount of tokens stored by the contract.
+     */
+    function totalSupply() external view returns (uint256);
+
+    /**
+     * @dev enumerate the owner token list.
+     * @param owner the address of token owner.
+     * @param index the index of owner token list.
+     * return a token ID owned by `owner` at a given `index` of its token list.
+     */
+    function tokenOfOwnerByIndex(address owner, uint256 index) external view returns (uint256 tokenId);
+
+    /**
+     * @dev get a tokenId through index of contract token list.
+     * @param index the index of contract token list.
+     * @return a tokenId at a given `index` of all the tokens stored by the contract.
+     */
+    function tokenByIndex(uint256 index) external view returns (uint256);
+}
+```
+
+#### 2. ERC-721
+
+Fully compatible with ERC-721
+
+```solidity
+interface ERC721{
+    event Transfer(address indexed _from, address indexed _to, uint256 indexed _tokenId);
+    event Approval(address indexed _owner, address indexed _approved, uint256 indexed _tokenId);
+    event ApprovalForAll(address indexed _owner, address indexed _operator, bool _approved);
+    function balanceOf(address _owner) external view returns (uint256);
+    function ownerOf(uint256 _tokenId) external view returns (address);
+    function safeTransferFrom(address _from, address _to, uint256 _tokenId, bytes data) external payable;
+    function safeTransferFrom(address _from, address _to, uint256 _tokenId) external payable;
+    function transferFrom(address _from, address _to, uint256 _tokenId) external payable;
+    function approve(address _approved, uint256 _tokenId) external payable;
+    function setApprovalForAll(address _operator, bool _approved) external;
+    function getApproved(uint256 _tokenId) external view returns (address);
+    function isApprovedForAll(address _owner, address _operator) external view returns (bool);
+}
+interface ERC165 {
+    function supportsInterface(bytes4 interfaceID) external view returns (bool);
+}
+```
+
+
+
+## Rationale
+
+#### 1 . Mortgage
+
+When mortgage NFT, you only need to mortgage the right to use into the mortgage contract, leaving the right to use.The advantage of this is that you can still use your own NFT before delivery.
+
+#### 2 . Rent 
+
+Lease does not require any collateral. First, you mortgage the NFT ownership into the contract, and the tenantry leases the right to use.
+
+This has the following benefits ：
+
+- No need to mortgage assets
+- Don't worry about the risk of default
```

Comment:
Similar to above, I'm personally not fully understanding the intended path to leverage this EIP for NFT mortgages or rentals. The `Rationale` section is more for design decisions, so I don't know if this would be the right place to put it, but I think that at least for myself, if there would be more in the `Motivation` explaining how this EIP is intended to be leveraged, it would help me understand here.

This would also be a good section for discussing why the authors opted to create a more simplified [EIP-2612](./eip-2612.md), and any other alternatives that were considered but ultimately discarded in favour of the current EIP.